src/pudl/output/sec10k.py

```diff
@@ -0,0 +1,108 @@
+"""Denormalized output tables for the SEC 10-K assets.
+
+These tables are created by joining the raw SEC 10-K tables with other data from the
+PUDL database, and enriching them with additional information. The resulting tables are
+more user-friendly and easier to work with than the normalized core tables.
+"""
+
+import dagster as dg
+import pandas as pd
+
+
+@dg.asset(
+    io_manager_key="pudl_io_manager",
+    group_name="out_sec10k",
+)
+def out_sec10k__quarterly_filings(
+    core_sec10k__quarterly_filings: pd.DataFrame,
+) -> pd.DataFrame:
+    """Denormalized table for SEC 10-K quarterly filings.
+
+    This table contains the basic information about the quarterly filings, including
+    the filing date, report date, and the URL to the filing.
+    """
+    # Construct the source URL so people can see where the data came from.
+    return core_sec10k__quarterly_filings.assign(
+        source_url=lambda x: (
+            "https://www.sec.gov/Archives/edgar/data/" + x["filename_sec10k"] + ".txt"
+        )
+    )
+
+
+@dg.asset(
+    io_manager_key="pudl_io_manager",
+    group_name="out_sec10k",
+)
+def out_sec10k__quarterly_company_information(
+    core_sec10k__quarterly_company_information: pd.DataFrame,
+    core_sec10k__quarterly_filings: pd.DataFrame,
+    core_sec10k__assn_sec10k_filers_and_eia_utilities: pd.DataFrame,
+    core_eia__entity_utilities: pd.DataFrame,
+) -> pd.DataFrame:
+    """Company information extracted from SEC10k filings and matched to EIA utilities."""
+    company_info = (
+        pd.merge(
+            left=core_sec10k__quarterly_company_information,
+            right=core_sec10k__assn_sec10k_filers_and_eia_utilities.loc[
+                :, ["central_index_key", "utility_id_eia"]
+            ],
+            how="left",
+            on="central_index_key",
+            validate="many_to_one",
+        )
+        .merge(
+            core_eia__entity_utilities.loc[:, ["utility_id_eia", "utility_name_eia"]],
+            how="left",
+            on="utility_id_eia",
+            validate="many_to_one",
+        )
+        .merge(
+            core_sec10k__quarterly_filings.loc[
+                :, ["filename_sec10k", "report_date", "filing_date"]
+            ],
+            how="left",
+            on="filename_sec10k",
+            validate="many_to_one",
+        )
+        .convert_dtypes()
+    )
+    company_info["source_url"] = (
+        "https://www.sec.gov/Archives/edgar/data/"
+        + company_info["filename_sec10k"]
+        + ".txt"
+    )
+    return company_info
+
+
+@dg.asset(
+    io_manager_key="pudl_io_manager",
+    group_name="out_sec10k",
+)
+def out_sec10k__changelog_company_name(
```

Comment:
i feel a little confused what's happening here - `company_name_old` and `company_name_new` indicate a name change on the `name_change_date`, and then `company_name` indicates the current name of the company (in the most recent filing)?

Reply:
`company_name` is ultimately dropped -- it's only here to provide the **current** company name associated with the CIK so we can fill in the most recent value for  `company_name_new` since only the former name is included in the changelog data.

Reply:
Ah got it, I was looking at a pickled version of the table which had `company_name` in it.